Rename Extension Headers to Properties (#1504)
moq-wg/moq-transport: draft-ietf-moq-transport.md

3. Use TRACK_STATUS or similar mechanism to query the previous state to
   determine the largest published Group ID.

> ## Properties {#Properties}

Author: ```suggestion
## Properties {#properties}
```

Tracks and Objects can have additional relay-visible fields, known as
Properties, which do not require negotiation, and can be used to alter
MoQT Object distribution.

Properties are defined in {{moqt-properties}} as well as external

> specifications and are registered in an IANA table {{iana}}. These

Reviewer: If #1473 lands, then Properties do not always need to be defined here or in a specification.

specifications define the type and value of the property, along with any rules
concerning processing, modification, caching and forwarding.

If unsupported by the relay, Properties MUST NOT be modified, MUST be

> cached as part of the Track or Object and MUST be forwarded by relays.  If a

Reviewer: This would read better in the singular.

## Relay Track Handling

A relay MUST include all Properties associated with a Track when sending any PUBLISH,
SUBSCRIBE_OK, REQUEST_OK when in response to a TRACK_STATUS, or FETCH_OK, unless
allowed by the property's specification (see {{properties}}).

## Relay Object Handling

> MOQT encodes the delivery information via Object properties ({{message-object}}).

Author: Yes, fixed.

Message Parameters are always intended for the peer endpoint only and are not
forwarded by Relays, though relays can consider received parameter values when
making a request.  Any Track metadata sent by the publisher that is forwarded to

> subscribers is sent as Track Properties.

Author: I rewrote this because I'm also trying not to use metadata if possible.

ID in the Track known by the subscriber, plus 1. A value of 0 indicates that the
subscriber has no Group information for the Track.  A subscriber MUST NOT send
this parameter in PUBLISH_OK or REQUEST_UPDATE if the Track did not

> include the DYNAMIC_GROUPS Parameter with value 1.  A subscriber MAY

Author: ```suggestion
include the DYNAMIC_GROUPS Property with value 1.  A subscriber MAY
```

specified track, as well as associated metadata required to deliver,
cache, and forward it.  Objects are sent by publishers.

### Object Header {#object-header}

> A canonical MoQ Object has the following information:

Author: Done

If omitted, the value is 0.

## Immutable Properties

> Immutable Extensions (Property Type 0xB) contains a sequence of

Author: ```suggestion
Immutable Properties (Property Type 0xB) contain a sequence of
```

}
~~~

This Property can be added by the Original Publisher, but MUST NOT be added by
Relays. This Property MUST NOT be modified or removed and the serialization
(e.g. variable-length integer encodings) of the Key-Value-Pairs MUST NOT
change). Like other Properties, Relays MUST cache Immutable Properties if the
Object or Track are cached and MUST forward it. Relays MAY decode and view
the Properties in the Key-Value-Pairs.

Unless specified by a particular Property specification, Properties
MAY appear either in the mutable extension list or inside Immutable Properties.
When looking for the value of a property, processors MUST search both the
mutable properties and the contents of Immutable Extensions.

If a Property allows multiple values, the same Property Type MAY appear in
both the mutable list and inside Immutable Extensions, unless prohibited by
the Property specification.

A Track is considered malformed (see {{malformed-tracks}}) if any of the
following conditions are detected:

>  * An Object contains an Immutable Properties header that contains another

Author: ```suggestion
 * An Object contains an Immutable Properties property that contains another
```
